internal/k8s/serialize.go

```diff
@@ -43,23 +44,47 @@ func decodeToRuntimeObj(ext runtime.RawExtension) (runtime.Object, error) {
 		return nil, nil
 	}
 
-	obj, _, err :=
+	obj, _, decodeErr :=
 		scheme.Codecs.UniversalDeserializer().Decode(ext.Raw, nil, nil)
-	if err == nil {
+	if decodeErr == nil {
 		return obj, nil
 	}
 
-	if !runtime.IsNotRegisteredError(err) {
+	// decode as unstructured - if the _original_ decode error was due to it
+	// being a non-standard type, the unstructured object will be returned;
+	// otherwise, it'll be used to provide additional context to the error if
+	// possible
+	var unst unstructured.Unstructured
+	_, gvk, err :=
+		unstructured.UnstructuredJSONScheme.Decode(ext.Raw, nil, &unst)
+	if err != nil {
+		if decodeErr != nil {
```

Comment:
this is guaranteed to be true after line 49, yeah?

Reply:
Yup 🤦 

Tightened this up in [8b9be03](https://github.com/tilt-dev/tilt/pull/5741/commits/8b9be03e8e759fd127466964241e82017e5be314)